examples/llm/src/models/mosaic_gpt/mosaic_gpt.py

```diff
@@ -72,6 +81,22 @@ def __init__(self, config: MosaicGPTConfig):
                     for _ in range(config.n_layers)
                 ])
         })
+        if config.tensor_parallel_mlp and is_torch_2_or_higher(
+        ) and enable_2d_with_fsdp():
```

Comment:
looks like right now if `enable_2d_with_fsdp` fails it will silently not apply. Could we break that out into a separate exception? (I don't know the context in which that would fail, so feel free to disregard if this doesn't make sense)

Reply:
will do and raise an exception.